packages/@sanity/cli/src/actions/init-project/initProject.ts

```diff
@@ -752,6 +778,59 @@ export default async function initSanity(
   function getOrganizationsWithAttachGrant(organizations: ProjectOrganization[]) {
     return pFilter(organizations, (org) => hasProjectAttachGrant(org.id), {concurrency: 3})
   }
+
+  async function createOrAppendEnvVars(filename: string) {
+    const variables = {
+      SANITY_PROJECT_ID: projectId,
+      SANITY_DATASET: datasetName,
+    }
+
+    try {
+      // use vercel's framework detector
+      const framework: Framework | null = await detectFrameworkRecord({
+        fs: new LocalFileSystemDetector(workDir),
+        frameworkList: frameworks as readonly Framework[],
+      })
+
+      if (framework && framework.slug !== 'sanity' && framework.envPrefix) {
+        print(
+          `\nDetected framework ${chalk.blue(framework?.name)}, using prefix '${
+            framework.envPrefix
+          }'`
+        )
+      }
+
+      await writeToEnv(filename, variables, {
+        framework,
+      })
+    } catch (err) {
+      print(err)
+      throw new Error('An error occurred while creating .env', {cause: err})
+    }
+  }
+
+  async function writeToEnv(
+    filename: string,
+    object: Record<string, string>,
+    options = {} as {
+      framework?: Framework | null
+    }
+  ) {
+    const prefix = options.framework?.slug === 'sanity' ? '' : options.framework?.envPrefix ?? ''
+    const envVariables = Object.keys(object)
+      .map((key) => `${prefix}${key}="${object[key]}"`)
+      .join('\n')
+
+    try {
+      // TO-DO: instead of appending if file exists, let's check whether variables are already set so we can just update them
+
+      // flag 'a' appends if exists
+      await fs.writeFile(filename, envVariables.concat('\n'), {flag: 'a', encoding: 'utf-8'})
```

Comment:
Good suggestion. It does add another manual step (prompt) to configure the output path, but despite that, I can still see it being useful.

For our new starter examples that are decoupled into `app` and `studio`, having support to provide an output path could allow one command to look like so:

`npm create sanity@latest --env --output-path app` &&
`npm create sanity@latest --env --output-path studio` to create env vars for `studio/.env`, `app/.env`.

Saying that, they would still have to go through the same flow and select the same project twice.

What about adding support for multiple dirs, such as `npm create sanity@latest --env --output-path app studio` to automatically write to both `app/` and `studio/` in the same command? @rexxars

Reply:
> It does add another manual step (prompt) to configure the output path, but despite that, I can still see it being useful.

I'm not sure if we need to prompt for an output dir, was mostly wondering if we should ignore the flag if specified, or respect it (eg prepend the env var file path with it, if present).

Not sure it's worth the extra complexity - if we did support multiple output paths, it would probably be by specifying the flag multiple times. Theoretically paths can contain commas, so seems a bit error prone to split on it. It would also have to error when outputting a studio and not an env file, since it doesn't make sense to output it to multiple places. My two cents.